### PR TITLE
feat:shifted gRPC call of cluster and repo creation to Argo REST api call

### PR DESCRIPTION
**Pull request by adi6859**
# Description

shifted gRPC call of cluster and repo creation to Argo REST api call
Fixes [AB2571](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=2571)
Fixes [AB2869](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=2869)

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
1.  Testing havebeen done locally
2. also tested by deploying image on my VM.
3. After shifting repository to argo rest api , done complete CI/CD and it is also working fine

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: internal/util/ArgoUtil/ClusterService.go [hunk 1]
```diff
@@ -21,28 +21,49 @@ import (
...
+	Name          string        `json:"name"`
+	ServerVersion string        `json:"serverVersion"`
+	Config        ClusterConfig `json:"config"`
+}
+
+type ArgoClusterService interface {
 	GetClusterByServer(server string) (*v1alpha1.Cluster, error)
 	ClusterList() (*v1alpha1.ClusterList, error)
 	CreateCluster(cluster v1alpha1.Cluster) (*v1alpha1.Cluster, error)
+	CreateClusterWithGitposConfigured(request *ClusterRequest, acdToken string) (*ClusterRequest, error)
```
Comment (iamayushm) [inline internal/util/ArgoUtil/ClusterService.go:48]: function name CreateClusterWithGitposConfigured doesn't make much sense, rename it

Comment (adi6859) [reply internal/util/ArgoUtil/ClusterService.go]: done

**Review by prakash100198 — APPROVED** [review]

File: internal/util/ArgoUtil/RepositoryService.go [hunk 2]
```diff
@@ -43,3 +50,11 @@ func (impl RepositoryServiceImpl) Create(repositoryRequest *v1alpha1.Repository)
 	}
 	return res, nil
 }
+func (impl RepositoryServiceImpl) CreateRepository(repositoryRequest *Repository, acdToken string) (repository *Repository, err error) {
+	res := &Repository{}
+	_, _, err = impl.DoRequestForArgo(&ClientRequest{ResponseBody: res, Path: impl.location, Method: "POST", RequestBody: repositoryRequest}, acdToken)
+	if err != nil {
+		return nil, err
```
Comment (vikramdevtron) [inline internal/util/ArgoUtil/RepositoryService.go:57]: log error

Comment (adi6859) [reply internal/util/ArgoUtil/RepositoryService.go:57]: done

**Pull request closed without merging**